sdlib/d/gc/region.d

```diff
@@ -234,25 +254,43 @@ unittest acquire_release {
 	void* addr0;
 	assert(regionAllocator.acquire(&addr0));
 
+	// check the block counter works
+	assert(regionAllocator.nBlocks == 1);
+
 	// Check we compute the proper range.
 	auto r = regionAllocator.computeAddressRange();
 	assert(!r.contains(addr0 - 1));
 	assert(r.contains(addr0));
 	assert(r.contains(addr0 + RefillBlockCount * BlockSize - 1));
 	assert(!r.contains(addr0 + RefillBlockCount * BlockSize));
 
+	bool passedThreshold = false;
 	foreach (i; 1 .. RefillBlockCount) {
 		void* addr;
 		assert(regionAllocator.acquire(&addr));
 		assert(addr is addr0 + i * BlockSize);
 		assert(r.contains(addr));
+		assert(regionAllocator.nBlocks == i + 1);
+		if (!passedThreshold && regionAllocator.checkBlockThreshold()) {
+			passedThreshold = true;
+			assert(regionAllocator.nBlocks == MinimumBlockCollectThreshold);
+		}
 	}
 
+	assert(regionAllocator.checkBlockThreshold());
+
 	foreach (i; 5 .. RefillBlockCount) {
 		void* addr = addr0 + i * BlockSize;
 		regionAllocator.release(addr, 1);
```

Comment:
You might want to check nBlocks here, as well as the loop at line 340 (github won't let me comment there...)

Reply:
I'm no longer testing the value tracker functionality, as that is now separate and tested separately.